### ISSUE TITLE
Library views and Random mix preference interaction

### ISSUE THREAD

**Issue by epoch1970** [LMS-Community/slimserver#531]
Hi Michael ;)
I think there is an issue when setting a player to use a library view, and at the same time setting random mix to pick songs from another library view.
In this case 'playlist add randomplay://tracks' returns an empty playlist.

I had an install running fine for a long time when a single library view was defined. Then when a second one was added, empty random playlists started to crop up.
Bug or not? Anything can be done?

Cheers.
https://forums.slimdevices.com/showthread.php?113861-Player-library-views-and-random-mix-preferences-interaction

**Comment by mherger** [comment]
Ok, I can confirm there's some conflict: the random play plugin would extract the correct track ID, but when trying to play, the `playlist addtracks` command would apply the client's library, filtering out that track ID. I'm not sure yet whether this should be considered a bug in the `addtracks` routine, which, given a track ID, should just play that track. Or whether `RandomPlay` should always define what library to use...

**Comment by epoch1970** [comment]
Me again...
I think I have found another ramification of the initial problem, where random selection can be locked-down to a specific genre that doesn't exist in the library view of a player. That player will cease receiving new items for its playlist.
I have _not_ verified the problem exists in the latest version of LMS.
I am able to work around the issue, poorly, by sending `playerid randomplaygenreselectall 0` right before sending `playerid playlist add randomplay://tracks`. But that doesn't stick if someone uses the GUI and selects genres while the player is active.

EDIT. I looked at Mixer.pm and found I could easily unset $queryGenres in case getLibraryIdForClient() has given a value to $queryLibrary. That's enough for me.
